RxSwiftTest/WheatherAPI.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  ApiController.swift
+//  RxSwiftTest
+//
+//  Created by Ran Helfer on 27/12/2021.
+//
+
+import Foundation
+import RxSwift
+
+class ApiController {
+    
+    static let shared = ApiController.init()
+    let apiKey = "ce9c6247ec5b529de72e47123fb3a403"
+    let baseUrlSting = "http://api.openweathermap.org/data/2.5/weather"
+    
+    func observableRequestForModel<T: Decodable>(request: URLRequest, emptyType: T) -> Observable<T> {
+        return URLSession.shared.rx.data(request: request).map({ data in
+            do {
+                let model = try JSONDecoder().decode(T.self, from: data)
+                return model
+            } catch let error {
+                return emptyType
+            }
+        })
+    }
```

Comment:
Dov, Elaz, i think we can use something like this in our app ? just need some polish and removal of `emptyType`

What do you think ?

Reply:
we have similar methods in ApolloClient+Reactive, using `Obsrvable.create ...`, which is even easier for creating observers - but I think we should discuss if new apis should still use rxSwift or move to combine, especially considering SwiftUI's extensive use of that.

Reply:
Thanks !  Sounds like a good point

Just started with `RxSwift` and it is no longer relevant... 🐰

Reply:
Combine is very similar, it's the exact same concepts, just with some added flavours.